react/features/polls/components/PollCreateForm.js

```diff
@@ -0,0 +1,131 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Container } from '../../base/react';
+import { FieldTextStateless } from '@atlaskit/field-text';
+import AddPollItem from './AddPollItem';
+import PollChoice from './PollChoice';
+import { translate } from '../../base/i18n';
+
+type Props = {
+
+    /**
+     * Available poll choices.
+     */
+    choices: Object,
+
+    /**
+     * Redux Store dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * Function handler to add a new choice.
+     */
+    onAddChoice: Function,
+
+    /**
+     * Function handler when poll choice text change.
+     */
+    onChoiceTextChange: Function,
+
+    /**
+     * Function handler to remove poll choice.
+     */
+    onRemoveChoice: Function,
+
+    /**
+     * Function handler when question text change.
+     */
+    onQuestionTextChange: Function,
+
+    /**
+     * The i18n translate function.
+     */
+    t: Function,
+
+    /**
+     * Current User ID.
+     */
+    userID: string,
+};
+
+/**
+ * Form for creating a new poll.
+ */
+class PollCreateForm extends Component<Props, *> {
+
+    /**
+     * Constructor.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._renderPollChoice = this._renderPollChoice.bind(this);
+    }
+
+    /**
+     * Component render method.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { onAddChoice, onQuestionTextChange, t } = this.props;
+        const choices = Object.keys(this.props.choices)
+            .map(this._renderPollChoice);
```

Comment:
What is the advantage of the structure of keeping poll choices separate from the poll object itself? I'm currently not seeing an advantage over nesting the poll choices within the poll. I understand it could be a good practice to treat redux like a database but I'm not convinced for this use case.

Reply:
Correct, normalizing (treating redux like a database with lookup keys) occurs usually when expecting or reaching a certain level of complexity of nested objects. I asked because I don't know if you've reached that point of complexity, but maybe you have.